lookout/style/format/utils.py

```diff
@@ -0,0 +1,28 @@
+from copy import deepcopy
+from typing import Mapping
+
+
+def merge_dicts(dict1: Mapping, dict2: Mapping) -> Mapping:
+    """
+    Deep merge of nested dictionaries.
+    Example:
+    >>> a = {1: 1, 2: {3: 3, 4: 4}}
+    >>> b = {1: 10, 2: {3: 30, 4: 4, 5: 5}}
+    >>> merge_dicts(a, b)
+    {1: 10, 2: {3: 30, 4: 4, 5: 5}}
+    >>> merge_dicts(b, a)
+    {1: 1, 2: {3: 3, 4: 4, 5: 5}}
+
+    :return: new merged dictionary
+    """
+
+    res = deepcopy(dict1)
+    stack = [(res, dict2)]
```

Comment:
Why not implement it as `dict(ChainMap(dict1, dict2))`?

Reply:
because it is not deep:
```
r = ChainMap(dict(a=dict(c=dict(d=1))),
             dict(a=dict(c=dict(d2=2))))
r["a"]["c"]["d"]
Out[10]: 1
r["a"]["c"]["d2"]
Traceback (most recent call last):
  File "/home/k/.local/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2910, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-11-22f82c69db02>", line 1, in <module>
    r["a"]["c"]["d2"]
KeyError: 'd2'

```

Reply:
You should ask in the python mailing list or issues or whatever they use if it makes sense to add `DeepChainMap`

Reply:
oookey, I means we cannot use it in 
https://github.com/src-d/style-analyzer/blob/3961fa9b5693e5c71a54e7aa0a771d190f103dfb/lookout/style/format/analyzer.py#L107
too

Reply:
they mean something else for `DeepChainMap` https://docs.python.org/3/library/collections.html#chainmap-examples-and-recipes,
ok, I will ask.